source/s3_client.h

```diff
@@ -0,0 +1,17 @@
+#ifndef AWS_CRT_PYTHON_S3_CLIENT_H
```

Comment:
trivial: maybe just name this file s3.h instead of s3_client.h
It's simpler to just have 1 header per submodule